tests-e2e/gappy_state_test.go

```diff
@@ -71,28 +79,50 @@ func TestGappyState(t *testing.T) {
 		})
 	}
 
-	t.Log("Alice requests an initial sliding sync on device 2.")
+	t.Logf("The proxy is now %d events behind the HS, which should trigger a limited sync", 1+numOtherState+numMessages)
+
+	t.Log("Alice requests an initial sliding sync on device 2, with timeline limit big enough to see her first message at the start of the test.")
 	syncResp = alice.SlidingSync(t,
 		sync3.Request{
 			Lists: map[string]sync3.RequestList{
 				"a": {
 					Ranges: [][2]int64{{0, 20}},
 					RoomSubscription: sync3.RoomSubscription{
-						TimelineLimit: 10,
+						TimelineLimit: 100,
```

Comment:
This test does produce a limited sync for the poller. But that poll is for a new device so gets timelimit limit 1 which wasn't satisfactory---I wanted to confirm that we include timeline events after the gap, but not before.

I felt it was easier to write an integration test so I can use the same poller without any logging out and in shenanigans.